test/test_xlib_display.py

```diff
@@ -0,0 +1,175 @@
+
+    def test_keycode_to_keysym_for_invalid_index(self):
+        self.assertEqual(self.display.keycode_to_keysym(0, 0), Xlib.X.NoSymbol)
+
+    def test_keysym_to_keycode_for_nosymbol(self):
+        self.assertEqual(self.display.keysym_to_keycode(Xlib.X.NoSymbol), 0)
+
+    def test_keysym_to_keycode_for_valid_symbol(self):
+        self.assertEqual(self.display.keysym_to_keycode(65535L), 119)
+
+    def test_keysym_to_keycodes_for_nosymbol(self):
+        self.assertEqual(self.display.keysym_to_keycodes(Xlib.X.NoSymbol), [])
+
+    def test_refresh_keyboard_mapping_invalid_event(self):
+        with self.assertRaises(TypeError) as e:
+            self.display.refresh_keyboard_mapping(Xlib.protocol.event.ResizeRequest())
+
+    def test_get_modifier_mapping(self):
+        self.assertEqual(len(self.display.get_modifier_mapping()), 8)
+
+    def test_set_modifier_mapping(self):
+        mapping = self.display.get_modifier_mapping()
+        self.assertEqual(self.display.set_modifier_mapping(mapping), Xlib.X.MappingSuccess)
+
+    def test_get_screensaver(self):
+        self.assertTrue(isinstance(self.display.get_screen_saver(), Xlib.protocol.request.GetScreenSaver))
+
+    def test_list_hosts(self):
+        self.assertTrue(isinstance(self.display.list_hosts(), Xlib.protocol.request.ListHosts))
+
+    def test_get_keyboard_control(self):
+        self.assertTrue(
+            isinstance(self.display.get_keyboard_control(), Xlib.protocol.request.GetKeyboardControl))
+
+    def test_change_keyboard_mapping(self):
+        kpt_mapping = self.display.get_keyboard_mapping(254, 1)
+        self.display.change_keyboard_mapping(254, kpt_mapping)
+        self.assertEqual(self.display.get_keyboard_mapping(254, 1), kpt_mapping)
+
+    def test_get_font_path(self):
+        self.assertNotEqual(self.display.get_font_path(), [])
+
+    def test_get_atom_name(self):
+        atom = self.display.get_atom(":99.0")
+        val = self.display.get_atom_name(atom)
+        self.assertEqual(val, ":99.0")
+
+    def test_intern_atom(self):
+        atom = self.display.intern_atom(":99.0")
+        val = self.display.get_atom_name(atom)
+        self.assertEqual(val, ":99.0")
+
+    def test_get_input_focus(self):
+        self.assertTrue(isinstance(self.display.get_input_focus(), Xlib.protocol.request.GetInputFocus))
+
+    def test_query_keymap(self):
+        self.assertTrue(isinstance(self.display.query_keymap(), list))
+
+    def test_open_invalid_font(self):
+        self.assertEqual(self.display.open_font("QQQ"), None)
+
+    def test_list_fonts(self):
+        fonts = self.display.list_fonts("*", 1)
+        self.assertNotEqual(fonts, [])
+
+    def test_lookup_valid_keysym(self):
+        self.assertNotEqual(self.display.lookup_string(65535L), None)
+
+    def test_lookup_invalid_keysym(self):
+        self.assertEqual(self.display.lookup_string(-1), None)
+
+    def test_rebind_string(self):
+        self.display.rebind_string(65535L, "qqq")
```

Comment:
`L` for integers is not supported in Py3.x. That's why Py3 tests are broken (syntax error in `test_xlib_display.py`). `__import__` call just doesn't output any details. But you may try to execute `test_xlib_display.py` only (not touching `testall.py`) and see which errors are here for Py3.